app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,66 @@
+class SubmissionsController < ApplicationController
+  layout "users"
+  before_action :current_user
+
+  def create
+    @submission = Submission.new(submission_params)
+    if @submission.save
+      undo_link = view_context.link_to('UNDO', circle_submission_path(circle_id: @submission.circle_id, id: @submission.id), method: :delete).html_safe
```

Comment:
Line is too long. [151/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.